### ISSUE TITLE
JSON::Object::set doesn't allow replacing values

### ISSUE THREAD

**Issue by erezeshkol** [pocoproject/poco#1138]
I'm using a JSONConfiguration in my program. A reasonable and common usage scenario for me is to update values in my configuration. However, Poco::JSON::Object::set uses map::insert to maintain its list of values. map::insert, in contrast to map::operator[], only adds and doesn't replace.
The result is that, when setting a new value to an existing key, the configuration isn't updated at all, with no notification of it.
Since I have no idea what the _preserveInsOrder clause does, I didn't want to change the functionality of the function and solved this by adding the following code before the line with insert:
``` 
    ValueMap::iterator existingValue = _values.find(key);
    if (existingValue != _values.end()) {
        _values.erase(existingValue);
    } 
```
A better solution would be to use operator[] and fix the function accordingly so whatever functionality is there in the _preserveInsOrder clause continues working.

**Comment by aleks-f** [comment]
Probably because of [this](https://github.com/pocoproject/poco/issues/933). Since entries are kept in a std::map, `_preserveInsOrder` ensures that stringification occurs in the original order. Without it, entries will be stringified sorted by keys.

**Comment by obiltschnig** [comment]
Should be fixed now through #933
